### PR TITLE
convert search file to typescript

### PR DESCRIPTION
**Pull request by AlexpdHackney**
### Description of change

Change the search page to Typescript.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -25,6 +25,6 @@
       "@/utils/*": ["src/utils/*"]
     }
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
+  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "src/components/Search/Search.test.tsx", "src/components/Search/Search.tsx"],
```
Comment (Duslerke) [inline tsconfig.json:28]: These 2 patterns seem like they are already part of the `"**/*.tsx"` pattern.

Comment (AlexpdHackney) [reply tsconfig.json]: @Duslerke This should be fixed now!

File: src/components/Search/Search.tsx [hunk 2]
```diff
@@ -10,10 +10,22 @@ import Meta from '../Meta'
...
-const Search = ({ query }) => {
+interface SearchProps {
+  query?: {
+    searchText?: string
+    pageNumber?: string
+  }
+}
+
+interface PropertyData {
+  total: string
```
Comment (Duslerke) [inline src/components/Search/Search.tsx:21]: Based on the line 90, I'm guessing it actually comes through as string from the API?
```js
setSearchHitTotal(parseInt(propertiesData.total))
```

File: src/components/Search/Search.tsx [hunk 3]
```diff
@@ -10,10 +10,22 @@ import Meta from '../Meta'
 import { canSearchForProperty } from '@/utils/userPermissions'
 import { PropertyListItem } from '@/models/propertyListItem'
 
-const Search = ({ query }) => {
+interface SearchProps {
+  query?: {
+    searchText?: string
+    pageNumber?: string
```
Comment (Duslerke) [inline src/components/Search/Search.tsx:16]: Guess it's the same issue with the pageNumber:
```ts
...(pageNumber && { pageNumber: parseInt(pageNumber) }),
```

**Review by Duslerke — APPROVED** [review]
Looking at the previous code, all is probably fine.

However, I raised the questions about the data types just in case.

File: src/components/Search/Search.tsx (hunk 2 above)
Comment (Duslerke) [reply src/components/Search/Search.tsx:21]: Hmm the API seems to be returning a `long` [[link to model](https://github.com/LBHackney-IT/repairs-api-dotnet/blob/423055144f1f7df6a8942c94f3d2c0906c0bca9b/RepairsApi/V2/Boundary/Response/SearchForPropertiesResponse.cs#L10)].

File: src/components/Search/Search.tsx (hunk 3 above)
Comment (Duslerke) [reply src/components/Search/Search.tsx:16]: The repairs API seems to treat this as an integer [[link to model](https://github.com/LBHackney-IT/repairs-api-dotnet/blob/423055144f1f7df6a8942c94f3d2c0906c0bca9b/RepairsApi/V2/Boundary/Request/SearchForPropertiesRequest.cs#L18)].

File: src/components/Search/Search.tsx (hunk 2 above)
Comment (LBHCallumM) [reply src/components/Search/Search.tsx:21]: its a number

**Review by LBHCallumM — APPROVED** [review]

**Pull request merged**